rclcpp/src/rclcpp/executor.cpp

```diff
@@ -275,7 +275,7 @@ Executor::spin_until_future_complete_impl(
   if (spinning.exchange(true)) {
     throw std::runtime_error("spin_until_future_complete() called while already spinning");
   }
-  RCPPUTILS_SCOPE_EXIT(this->spinning.store(false); );
+  RCPPUTILS_SCOPE_EXIT(this->spinning.store(false);wait_result_.reset(););
```

Comment:
i guess this is the safest way to make sure entities are released after spin, even this could be redundant call for the spin loop. if the executor is static global in process space, user would shut down the program after spin? correct?

Reply:
Yes.